classes/converter_chromium.php

```diff
@@ -139,6 +183,20 @@ protected function generate_pdf_content(moodle_url $proxyurl, string $filename =
         }
     }
 
+    /**
+     * Filters the provided options array to remove rendering-specific options, leaving only PDF-specific options.
+     *
+     * @param array $options any additional options to pass to converter, valid options vary with converter
+     * instance, see relevant converter for further details.
+     * @return array The filtered array, containing only the options relevant for PDF generation.
+     */
+    protected function filter_pdf_options(array $options) {
+        return array_filter($options, function($option) {
+            $renderoptions = ['windowSize', 'userAgent', 'jsCondition', 'jsconditionparams', 'customFlags'];
+            return !in_array($option, $renderoptions);
+        }, ARRAY_FILTER_USE_KEY);
```

Comment:
Optional thing (don't feel like you need to do it): Now might be a good time to rewrite this:

```
$renderoptions = ['windowSize', 'userAgent', 'jsCondition', 'jsconditionparams', 'customFlags'];
return array_diff_key($options, array_flip($renderoptions));
```

Reply:
Also let's add `array` return type to the method signature.

Reply:
That looks so much better. Thanks for the suggestion. I've updated the code :)